### ISSUE TITLE
null_response not working on https with sniffing ON

### ISSUE THREAD

**Issue by iamwiliam** [luminati-io/luminati-proxy#25]
To the next url:
`https://www.facebook.com/fr/` (with HTTPS)

With the null_pattern:
`facebook`

It doesn't returns NULL because _url is "/fr/" in the next line:
` if (_this.null_response && _this.null_response.test(_url))` 

When you try with:
`http://www.facebook.com/fr/` (with HTTP)

The variable _url is "http://www.facebook.com/fr/" and not only the path "/fr/"

```
    if (req.method=='CONNECT')
    {
        let parts = _url.split(':');
        if (parts[0].match(/^\d+\.\d+\.\d+\.\d+$/) && _this.opt.resolve)
            parts[0] = yield _this.opt.resolve(parts[0])||parts[0];
        _url = parts.join(':');
        if (parts[0].match(/^\d+\.\d+\.\d+\.\d+$/))
            _this._log.warn('HTTPS connection to IP: ', _url);
    }

    _this._log.info(`${req.method} ${_url}`);

    if (_this.null_response && _this.null_response.test(_url))
    {
        _this._log.debug(`Returning null response: ${req.method} ${_url}`);
        let status = req.method=='CONNECT' ? 501 : 200;
        write_http_reply(res, {statusCode: status, statusMessage: 'NULL'});
        res.end();
        return;
    }
```

**Comment by lee-elenbaas** [comment]
Will be fixed in the next release

**Comment by lee-elenbaas** [comment]
fixed on 0.8.36

**Comment by iamwiliam** [comment]
Thanks, I made a fork to fix that temporarily, so now I can checkout HEAD again.
